src/JobsManager.cpp

```diff
@@ -0,0 +1,47 @@
+#include "JobsManager.h"
+#include <iostream>
+#include <sys/wait.h>
+
+
+/*
+* show the jobs that are currently running.
+* display the PID, command and status of each job.
+* kill jobs that have finished running.
+*/
+void JobsManager::showBackgroundJobs() const
+{
+    for (const auto& job : m_jobs) {
+        std::cout << "PID: " << job.first << ", Command: "
+            << job.second << ", Status: Running\n";
+    }
+}
+
+
+/*
+* clean up the jobs that have finished running.
+*/
+void JobsManager::cleanUpBackgroundJobs()
+{
+    for (auto it = m_jobs.cbegin(); it != m_jobs.cend();)
```

Comment:
Using a for loop without an increment step in the loop definition is dangerous, it might cause infinite loops. 
You should avoid incrementing in the for loop body. Actually it looks like you're implementing a while loop with a for loop